### PR TITLE
Revert "Dispatch upgradeAppStoreApps event"

### PR DESCRIPTION
**Pull request by PVince81**
Reverts owncloud/core#27648 and some cleanup for upgrade process.

As suggested in https://github.com/owncloud/core/pull/27711#issuecomment-296294887

@VicDeo

### CODE REVIEW

**Comment by PVince81** [comment]
Hmm, might as well cleanup the remaining unneeded code.

File: lib/private/Updater.php [hunk 1]
```diff
@@ -344,9 +344,7 @@ private function checkAppsRequirements() {
 			$info = OC_App::getAppInfo($app);
 			if(!OC_App::isAppCompatible($version, $info)) {
 				OC_App::disable($app);
-				$disabledApps[]= $app;
 				$this->emit('\OC\Updater', 'incompatibleAppDisabled', [$app]);
-				continue;
```
Comment (PVince81) [inline lib/private/Updater.php]: @VicDeo should we still keep this ? I'm not sure if a full revert is a good idea.

**Comment by PVince81** [comment]
If we revert this it seems some of the workflow will change, so might need a full retest of the update code.

File: lib/private/Updater.php (hunk 1 above)
Comment (VicDeo) [inline lib/private/Updater.php]: @PVince81  yes, looks like this part had a logic flaw before

**Comment by PVince81** [comment]
@VicDeo can you finish this for 10.0.1 ? Seems we'll need to keep some logic.

**Comment by PVince81** [comment]
@VicDeo please finish this if applicable, else close

**Comment by VicDeo** [comment]
@PVince81 @DeepDiver1975 do we need audisable as a last line of defence?
If not, I would just kill disabling/reenabling here at once

**Comment by PVince81** [comment]
UnresolvableWord: `audisable`

**Comment by VicDeo** [comment]
@PVince81 oh. it should sound like `auto-disable`

**Comment by PVince81** [comment]
@VicDeo auto-disable on update ? I think we removed that... currently the only auto-disable that's left is if the app throws an exception in app.php, and that's during app load, not install

**Comment by VicDeo** [comment]
@PVince81 not exactly. 
We removed disabling of all non-shipped apps.
While incompatible apps are still disabled here.

Another story that reaching this point is blocked by repair step.

**Comment by PVince81** [comment]
ah yes... incompatible apps... yes, let's disable these automatically

**Comment by PVince81** [comment]
moving to "planned"

**Comment by VicDeo** [comment]
Rebased.
Lines 345-356 looks pretty useless for me after a rebase

**Comment by VicDeo** [comment]
@PVince81 I'm feeling like killing more code here.
Incompatibility check has been moved into before upgrade repair step. So at least `checkAppsRequirements` is just a useless crap. There is no need in such duplication

**Comment by PVince81** [comment]
@VicDeo sounds good

**Comment by VicDeo** [comment]
Rebased. Squashed.

**Comment by PVince81** [comment]
:+1: apps are still properly fetched from the market

**Comment by PVince81** [comment]
@VicDeo please backport to stable10

**Comment by VicDeo** [comment]
Stable10: https://github.com/owncloud/core/pull/29249

**Pull request merged**